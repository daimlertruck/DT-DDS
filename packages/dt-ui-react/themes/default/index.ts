import { Theme } from '../..';

import {
  palette as alphaPalette,
  colors as alphaColors,
} from './alpha/palette';
import { fontStyles as alphaFontStyles } from './alpha/typography';
import { animations } from './animations';
import { breakpoints } from './breakpoints';
import { palette, colors } from './palette';
import { radius } from './radius';
import { shadows } from './shadows';
import { shape } from './shape';
import { spacing } from './spacing';
import { fontFamily, fontStyles, responsiveFontStyles } from './typography';

export const theme: Theme = {
  animations,
  fontFamily,
  fontStyles: {
    alpha: alphaFontStyles,
    ...fontStyles,
  },
  palette: {
    alpha: alphaPalette,
    ...palette,
  },
  colors: {
    alpha: alphaColors,
    ...colors,
  },
  radius,
  shadows,
  shape,
  spacing,
  breakpoints,
  responsiveFontStyles,
  icons: 'outlined',
};
