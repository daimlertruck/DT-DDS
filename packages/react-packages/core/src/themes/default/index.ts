import { CustomTheme } from '../../types';

import { animations } from './animations';
import { breakpoints } from './breakpoints';
import { palette, colors } from './palette';
import { radius } from './radius';
import { shadows } from './shadows';
import { shape } from './shape';
import { spacing } from './spacing';
import { fontFamily, fontStyles, responsiveFontStyles } from './typography';

export const theme: CustomTheme = {
  animations,
  breakpoints,
  palette,
  colors,
  radius,
  shadows,
  spacing,
  fontFamily,
  fontStyles,
  responsiveFontStyles,
  shape,
  icons: 'outlined',
};
