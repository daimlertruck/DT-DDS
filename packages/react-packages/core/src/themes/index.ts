export { theme } from './default';
export { globalStyles } from './globalStyles';
export { truckAPITheme } from './truckAPI';
