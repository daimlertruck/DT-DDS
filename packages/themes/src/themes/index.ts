export { default as defaultTheme } from './daimler-truck';
