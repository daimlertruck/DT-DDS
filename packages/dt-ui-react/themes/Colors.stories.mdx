import { Meta, ColorPalette, ColorItem } from '@storybook/addon-docs';

import { themes, ThemeProvider, convert, ensure } from '@storybook/theming';
import { theme } from './default';

<Meta title='Foundations/Colors' />

# Default

export const colors = Object.entries(theme.colors).reduce(
  (acc, [key, value]) => {
    const groupedKey =
      key.indexOf('_') > 0 ? key.substring(0, key.indexOf('_')) : key;
    const group = acc[groupedKey] ?? [];
    return {
      ...acc,
      [groupedKey]: [
        ...group,
        {
          [key]: value,
        },
      ],
    };
  },
  {}
);

<div style={{ background: '#eeeeee', padding: '20px' }}>
  <ThemeProvider theme={theme}>
    <ColorPalette>
      {Object.entries(colors).map(([k, v]) => {
        return <ColorItem key={k} title={k} colors={Object.assign({}, ...v)} />;
      })}
    </ColorPalette>
  </ThemeProvider>
</div>
