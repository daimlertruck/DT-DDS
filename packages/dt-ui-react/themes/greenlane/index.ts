import { breakpoints } from '../default/breakpoints';
import { colors } from '../default/palette';
import { radius } from '../default/radius';
import { shadows } from '../default/shadows';
import { spacing } from '../default/spacing';
import { responsiveFontStyles } from '../default/typography';

import { palette } from './palette';
import { fontFamily, fontStyles } from './typography';

export const greenlaneTheme = {
  fontFamily,
  fontStyles,
  palette,
  colors,
  shadows,
  spacing,
  breakpoints,
  responsiveFontStyles,
  radius,
};
