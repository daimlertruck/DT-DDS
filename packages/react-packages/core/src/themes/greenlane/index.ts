import { CustomTheme } from '../../types';
import { theme } from '../default/index';

import { palette } from './palette';
import { shape } from './shape';
import { fontFamily, fontStyles } from './typography';

export const greenlaneTheme: CustomTheme = {
  ...theme,
  fontFamily,
  fontStyles,
  palette,
  shape,
  icons: 'rounded',
};
