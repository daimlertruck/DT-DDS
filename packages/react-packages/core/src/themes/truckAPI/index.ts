import { CustomTheme } from '../../types';
import { theme } from '../default/index';

import { palette } from './palette';
import { shape } from './shape';
import { fontFamily, fontStyles } from './typography';

export const truckAPITheme: CustomTheme = {
  ...theme,
  fontFamily,
  fontStyles,
  palette,
  shape,
  icons: 'outlined',
};
