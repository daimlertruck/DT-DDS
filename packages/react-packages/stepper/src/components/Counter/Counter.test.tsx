import { withProviders } from '@dt-dds/react-core';
import { Colors, defaultTheme as theme } from '@dt-dds/themes';
import { render, screen } from '@testing-library/react';

import { Counter } from '.';

describe('<Counter /> component', () => {
  const ProvidedCounter = withProviders(Counter);
  it('should render a span html element with the given content', () => {
    const { container } = render(<ProvidedCounter>Active</ProvidedCounter>);

    expect(container).toMatchSnapshot();
  });

  it('should render properties of a disabled Counter', () => {
    render(<ProvidedCounter color='disabled'>Active</ProvidedCounter>);

    const counterEl = screen.getByTestId('counter');

    expect(counterEl).toHaveStyleRule('color', '#ffffff');
    expect(counterEl).toHaveStyleRule(
      'background-color',
      theme.palette.content.default
    );
  });

  it.each`
    isLarge  | expectedSize
    ${true}  | ${'24px'}
    ${false} | ${'16px'}
  `(
    'should have width & height equal to $expectedSize when isLarge is $isLarge ',
    ({ isLarge, expectedSize }: { isLarge: boolean; expectedSize: string }) => {
      render(<ProvidedCounter isLarge={isLarge}>1</ProvidedCounter>);

      const counterEl = screen.getByTestId('counter');

      expect(counterEl).toHaveStyleRule('width', expectedSize);
      expect(counterEl).toHaveStyleRule('height', expectedSize);
    }
  );

  it.each`
    outlined | expectedBorder                                      | expectedBackgroundColor
    ${true}  | ${`2px solid ${theme.palette.informative.default}`} | ${'#ffffff'}
    ${false} | ${'0px'}                                            | ${theme.palette.informative.default}
  `(
    'should have border and background equals to $expectedBorder, $expectedBackgroundColor when outlined is $outlined ',
    ({
      outlined,
      expectedBorder,
      expectedBackgroundColor,
    }: {
      outlined: boolean;
      expectedBorder: string;
      expectedBackgroundColor: string;
    }) => {
      render(<ProvidedCounter outlined={outlined}>1</ProvidedCounter>);

      const counterEl = screen.getByTestId('counter');

      expect(counterEl).toHaveStyleRule('border', expectedBorder);
      expect(counterEl).toHaveStyleRule(
        'background-color',
        expectedBackgroundColor
      );
    }
  );

  it.each`
    color            | expectedColor
    ${'disabled'}    | ${theme.palette.content.light}
    ${'informative'} | ${theme.palette.informative.default}
  `(
    'should have border and color equals to $expectedColor when color is $color and outlined is true ',
    ({
      color,
      expectedColor,
    }: {
      color: Colors | 'disabled';
      expectedColor: string;
    }) => {
      render(
        <ProvidedCounter color={color} outlined>
          2
        </ProvidedCounter>
      );

      const counterEl = screen.getByTestId('counter');

      expect(counterEl).toHaveStyleRule('border', `2px solid ${expectedColor}`);
      expect(counterEl).toHaveStyleRule('color', expectedColor);
    }
  );
});
