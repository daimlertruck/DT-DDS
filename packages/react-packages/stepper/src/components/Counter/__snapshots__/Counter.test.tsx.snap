// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> component should render a span html element with the given content 1`] = `
.emotion-0 {
  font-family: Daimler CS;
  font-size: 12px;
  font-weight: 400;
  letter-spacing: 0px;
  line-height: 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #ffffff;
  border: 0px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 16px;
  height: 16px;
  border-radius: 50%;
  background-color: #00677f;
}

<div>
  <span
    class="emotion-0 emotion-1"
    color="informative"
    data-testid="counter"
  >
    Active
  </span>
</div>
`;
